state/dbstore.go

```diff
@@ -34,6 +34,7 @@ type Store interface {
 	Get(key string, i interface{}) (err error)
 	Put(key string, i interface{}) (err error)
 	Delete(key string) (err error)
+	Keys() (keys []string, err error)
```

Comment:
The purpose of this method is to obtain a set of keys which match the given `prefix`.

The reason this is needed is that #1447 defines a feature which gets all of balances for a running node. For this, we must check not only memory, but the `leveldb` as well.

However, we use this db not only for balances, but for cheques as well. Indeed, rather than doing a full database scan, we care only about the balance entries for this functionality.

In order to get this, we need every entry named with the balances prefix (i.e. `balances_`).

This is the idea behind the `Keys` function.

Reply:
This has been addressed with PR https://github.com/ethersphere/swarm/pull/1629